packages/server/graphql/mutations/setAppLocation.ts

```diff
@@ -22,27 +24,39 @@ export default {
   })(async (_source, {location}, {authToken, dataLoader, socketId: mutatorId}) => {
     const operationId = dataLoader.share()
     const subOptions = {mutatorId, operationId}
+    const redis = getRedis()
 
     // AUTH
     const viewerId = getUserId(authToken)
     const viewer = viewerId ? await dataLoader.get('users').load(viewerId) : null
     if (!viewer) {
       return {error: {message: 'Not a user'}}
     }
+    const userPresence = await redis.lrange(`presence:${viewerId}`, 0, -1)
+    const connectedSocket = userPresence.find((socket) => JSON.parse(socket).socketId === mutatorId)
+    if (!connectedSocket) {
+      return {error: {message: "Socket doesn't exist"}}
+    }
 
     // RESOLUTION
-    const {lastSeenAtURL} = viewer
-    const lastSeenAt = new Date()
+    const parsedConnectedSocket = JSON.parse(connectedSocket) as UserPresence
+    const {lastSeenAtURL} = parsedConnectedSocket
+    const {lastSeenAt} = viewer
+    const now = new Date()
+    const datesAreOnSameDay = now.toDateString() === lastSeenAt.toDateString()
```

Comment:
i like this thought!
i think doing this once in `connectSocket` probably suffices.